WalletWasabi/JsonConverters/Bitcoin/CoinJsonConverter.cs.cs

```diff
@@ -0,0 +1,38 @@
+using NBitcoin;
+using Newtonsoft.Json;
+
+namespace WalletWasabi.JsonConverters.Bitcoin;
+
+public class CoinJsonConverter : JsonConverter<Coin>
+{
+	/// <inheritdoc />
+	public override Coin? ReadJson(JsonReader reader, Type objectType, Coin? existingValue, bool hasExistingValue, JsonSerializer serializer)
+	{
+		var coin = serializer.Deserialize<SerializableCoin>(reader);
+		return new Coin(coin.Outpoint, coin.TxOut);
+	}
+
+	/// <inheritdoc />
+	public override void WriteJson(JsonWriter writer, Coin? coin, JsonSerializer serializer)
+	{
+		if (coin is null)
+		{
+			throw new ArgumentNullException(nameof(coin));
+		}
```

Comment:
Why do we let `coin` parameter to be `null` if we throw an expection right away?

Reply:
Yes, exactly, we weren't the ones who designed this contract.